client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java

```diff
@@ -103,6 +104,26 @@ public void testSimpleQueryWithTSV() {
             Assert.fail("failed to get response", e);
         } catch (IOException e) {
             Assert.fail("failed to read response", e);
+        }
+    }
+
+    @Test(groups = {"integration"})
+    public void testReadQuery() {
+        prepareSimpleDataSet();
+
+        Future<QueryResponseReader> response = client.readQuery("SELECT * FROM " + DATASET_TABLE);
+
+        try {
+            ClickHouseBinaryFormatReader reader = response.get().getReader();

```

Comment:
why not have a records method?

Reply:
Agree. I will think and implement it more natural way as we discussed.